src/main/java/org/codehaus/plexus/archiver/jar/JarArchiver.java

```diff
@@ -435,7 +435,7 @@ private void createIndexList( ConcurrentJarCreator zOut )
         ByteArrayInputStream bais = new ByteArrayInputStream( baos.toByteArray() );
 
         super.zipFile( createInputStreamSupplier( bais ), zOut, INDEX_NAME, System.currentTimeMillis(), null,
-                       DEFAULT_FILE_MODE, null );
+                       DEFAULT_FILE_MODE, null, true );
```

Comment:
Not sure if the index is ok to be added in parallel. But it's seems this is the current behavior so I just kept it this way.